tests/unit/voucher/AccountAuthorized.test.ts

```diff
@@ -0,0 +1,124 @@
+import { Address, BigInt } from '@graphprotocol/graph-ts';
+import { assert, beforeEach, clearStore, describe, test } from 'matchstick-as/assembly/index';
+import { handleAccountAuthorized } from '../../../src/voucher';
+import {
+    createAccountAuthorizedEvent,
+    createAndSaveType,
+    createAndSaveVoucher,
+} from '../utils/utils';
+
+// Shared constants
+const VOUCHER_TYPE_ID = '1';
+const VOUCHER_DESCRIPTION = 'Test Voucher Type';
+const VOUCHER_DURATION = BigInt.fromI32(86400);
+const VOUCHER_TYPE_ELIGIBLE_ASSETS: string[] = [];
+
+const VOUCHER_OWNER = '0xabcdefabcdefabcdefabcdefabcdefabcdefabcd';
+const VOUCHER_VALUE = BigInt.fromI32(100);
+const VOUCHER_BALANCE = BigInt.fromI32(50);
+const VOUCHER_EXPIRATION = BigInt.fromI32(999999);
+
+describe('AccountAuthorizedEvent', () => {
+    beforeEach(() => {
+        clearStore();
+
+        // Initialize a VoucherType entity
+        createAndSaveType(
+            VOUCHER_TYPE_ID,
+            VOUCHER_DESCRIPTION,
+            VOUCHER_DURATION,
+            VOUCHER_TYPE_ELIGIBLE_ASSETS,
+        );
+    });
+
+    test('Should add an account to authorizedAccounts when the Voucher exists', () => {
+        // --- GIVEN
+        const voucherAddress = '0x1234567890123456789012345678901234567890';
+        const authorizedAccount = '0xabcdefabcdefabcdefabcdefabcdefabcdefabcd';
```

Comment:
To avoid possible confusion, It's better to use an address that is different from the owner address because in the Voucher contract it's not possible to authorize the owner address.
https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/blob/main/contracts/beacon/Voucher.sol#L359